### ISSUE TITLE
Host callback attempts to perform computation on GPU

### ISSUE THREAD

**Issue by mfschubert** [google/jax#13046]
### Description

I am trying to run a computation including `jax.numpy.linalg.eig` on a GPU.

Since `eig` is not implemented with GPU backend, I am trying to use the experimental `host_callback` module to force the calculation to take place on the CPU. Unfortunately, it seems like an attempt is still made to run this on the GPU, which fails.

This code snippet reproduces the error, for example in colab with a GPU runtime.

```
def _eig_host(matrix: jnp.ndarray) -> Tuple[jnp.ndarray, jnp.ndarray]:
    """Performs an `eig` solve on the host (CPU)."""
    eigenvalues_shape = jax.ShapeDtypeStruct(matrix.shape[:-1], complex)
    eigenvectors_shape = jax.ShapeDtypeStruct(matrix.shape, complex)
    return hcb.call(
        jnp.linalg.eig,
        matrix.astype(complex),
        result_shape=(eigenvalues_shape, eigenvectors_shape),
    )

x = jax.random.normal(jax.random.PRNGKey(0), (3, 3))
_eig_host(x)  # This fails, because `eig` is not implemented for GPU.
```

### What jax/jaxlib version are you using?

0.3.23

### Which accelerator(s) are you using?

colab GPU

### Additional system info

colab

**Comment by PhilipVinc** [comment]
the callback gives you CPU-bound numpy objects, so I think the ideal thing here would be to callback into `np.linalg.eig`.

**Comment by mfschubert** [comment]
Yes, this is what I originally intended, but for my version of jax and numpy, the jax implementation was much faster. So, I wanted to find a way to use the jax implementation.

**Comment by selamw1** [comment]
Update:
`jax.numpy.linalg.eig` is now supported on GPUs with JAX version [0.4.36](https://github.com/jax-ml/jax/releases/tag/jax-v0.4.36). See [#24663](https://github.com/jax-ml/jax/pull/24663) for more details.
